lib/services/elasticsearch.js

```diff
@@ -687,6 +687,62 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+  * Updates all documents matching the provided filters
+  *
+  * @param {String} index - Index name
+  * @param {String} collection - Collection name
+  * @param {Object} query - Query to match documents
+  * @param {Object} changes - Changes wanted on documents
+  * @param {Object} options - from (undefined), size (undefined), refresh (undefined), timeout (undefined), userId (undefined)
+  *
+  * @returns {Promise.<Object>} { documents, total, failures: [ _shardId, reason ] }
+  */
+  async updateByQuery(
+    index,
+    collection,
+    query,
+    changes,
+    { refresh, from, size } = {})
+  {
+
+    try {
+      const esRequest = {
+        index: this._getESIndex(index, collection),
+        body: this._sanitizeSearchBody({ query }),
+        scroll: '5s',
+        from,
+        size
+      };
+
+      const documents = await this._getAllDocumentsFromQuery(esRequest);
+      documents.forEach(document => {
+        document._source = null;
```

Comment:
`undefined` mean no value and `null` mean no object value, it's better to set the value to `undefined` when you want to get ride of it

```suggestion
        document._source = undefined
```

Reply:
Sonarqube does not allow it

Reply:
Rule deactivated